### DISCUSSION TITLE
Solcast param "hours"

### DISCUSSION THREAD

**Discussion by rpsmonkey** [StefaE/PVForecast#13] [Ideas]
Hi, 

thanks for the great work here!

It seems Solcast has changed the behaviour of the api. They now have the parameter "hours" and the one is by default set to 48 hours. So using this script I only get the forecast for the next 48 hours. The parameter is optional and can be set between 1 to 336. See Solcast Api documentation: https://docs.solcast.com.au/#58ca9bc0-27d4-4418-937f-03986331f01d

Unfortunately, I'm bad at python and could need some help implementing this parameter to the script. I'm also not sure if the used pysolcast (https://github.com/mcaulifn/solcast) needs to be changed?

What did I try?
In solecast.py I changed/added:
   self.postDict      = { 'hours':336 }                                                     # dictionary to post to solcat
   self.params = {'hours':338 }
Did not work for me. 

What did work:
I copied the classes and definition from pysolcast into the file solecast.py and changed this part:
>  def _get_data(self, uri: str, params: dict = None) -> dict:  # pylint: disable=inconsistent-return-statements
>  """Get data from API."""
>  
>   url = f'{PySolcast.base_url}{uri}'
>        payload = {'format': 'json'}
>        if params:
>            payload = {**payload, **params}`

to this:
>  def _get_data(self, uri: str, params: dict = None) -> dict:  # pylint: disable=inconsistent-return-statements
>        """Get data from API."""
>        url = f'{PySolcast.base_url}{uri}'
>        payload = {'format': 'json'}
>        **params = { 'hours':336 }**
>        if params:
>            payload = {**payload, **params}`
This works for me - but is not nice code..

Can someone help me understanding what I have to change in solecast.py so that I can give the params parameter when calling the function?

Kind Regards
Robert

**Comment by StefaE** [comment]
Thx., Robert for the detailed research. For my part, I was just surprised that the forecast period got reduced :-)

I raised an [issue](https://github.com/mcaulifn/solcast/issues/90) over at `pysolcast` to allow additional parameter passing. If this gets resolved, we can implement a clean solution. (Else, I need make PVForecasts self-sufficient and remove the dependency on pysolcast - but that'll take a while)

If and when we have the fix in pysolcast, we can write in `solcast.py.getSolCast`:
```
forecasts_1 = self._site.get_forecasts_parsed({'hours':self.hours})
if self._site_2 is not None: 
   forecasts_2 = self._site_2.get_forecasts_parsed({'hours':self.hours})
```
and take `self.hours` from the config file

**Reply by mcaulifn** [reply]
This was fixed in https://github.com/mcaulifn/solcast/releases/tag/v1.0.12

**Reply by StefaE** [reply]
Finally, I updated my part too ... see v2.01
